lib/model.js

```diff
@@ -622,7 +622,20 @@ comparisonFunctions.$size = function (obj, value) {
 
     return (obj.length == value);
 };
+comparisonFunctions.$elemMatch = function (obj, value) {
+  if (!util.isArray(obj)) { return false; }
+  var i = obj.length;
+  var result = false;   // Initialize result
+  while (i--) {
```

Comment:
I hadn't seen this way of doing it before. While it is smart I think it could mislead quite a number of dev, and would use the more common `for (i = 0; i < obj.length; i += 1) { ... }`.